lib/manager/bundler/extract.ts

```diff
@@ -38,33 +38,18 @@ export async function extractPackageFile(
     if (rubyMatch) {
       res.compatibility = { ruby: rubyMatch[1] };
     }
-    let gemMatch: RegExpMatchArray;
-    let gemDelimiter: string;
-    for (const delimiter of delimiters) {
-      const gemMatchRegex = `^gem ${delimiter}([^${delimiter}]+)${delimiter}(,\\s+${delimiter}([^${delimiter}]+)${delimiter}){0,2}`;
-      if (regEx(gemMatchRegex).test(line)) {
-        gemDelimiter = delimiter;
-        gemMatch = gemMatch || regEx(gemMatchRegex).exec(line);
-      }
-    }
+    const gemMatchRegex = /^\s*gem\s+(['"])(?<depName>[^'"]+)\1(\s*,\s*(?<currentValue>(['"])[^'"]+\5(\s*,\s*\5[^'"]+\5)?))?/;
+    const gemMatch = gemMatchRegex.exec(line);
     if (gemMatch) {
       const dep: PackageDependency = {
-        depName: gemMatch[1],
+        depName: gemMatch.groups.depName,
         managerData: { lineNumber },
       };
-      if (gemMatch[3]) {
-        let currentValue = gemMatch[0]
-          .substring(`gem ${gemDelimiter}${dep.depName}${gemDelimiter},`.length)
-          .trim();
-        // strip quotes unless it's a complex constraint
-        if (
-          currentValue.startsWith(gemDelimiter) &&
-          currentValue.endsWith(gemDelimiter) &&
-          currentValue.split(gemDelimiter).length === 3
-        ) {
-          currentValue = currentValue.slice(1, -1);
-        }
-        dep.currentValue = currentValue;
+      if (gemMatch.groups.currentValue) {
+        const currentValue = gemMatch.groups.currentValue;
+        dep.currentValue = /\s*,\s*/.test(currentValue)
+          ? currentValue
+          : currentValue.slice(1, -1);
```

Comment:
Preserve current behavior of range extraction, if needed: 
`gem "listen", ">= 3.0.5", "< 3.2", require: false` -> `currentValue === '">= 3.0.5", "< 3.2"'`